lib/value/src/value/object.rs

```diff
@@ -0,0 +1,168 @@
+//! Ordered associated object, mapping from string to `Value`.
+
+use std::{
+    cmp::Ordering,
+    collections::{btree_map, BTreeMap},
+    ops::{Deref, DerefMut},
+};
+
+#[cfg(feature = "serde")]
+use serde::{Serialize, Serializer};
+
+/// Wraps a string-keyed `BTreeMap`, similar to `ValueRgex` in purpose.
+#[derive(Debug, Clone)]
+#[repr(transparent)]
+pub struct Object<V>(BTreeMap<String, V>);
+
+impl<V> Object<V> {
```

Comment:
Hrm. That we don't have a `get` or `remove` here suggests the conversion away from `BTreeMap` isn't totally complete yet. I don't _think_ that our value objects are append-only.